framework/src/auxkernels/AuxKernel.C

```diff
@@ -108,6 +111,12 @@ AuxKernel::AuxKernel(const InputParameters & parameters) :
   for (const auto & it : coupled_vars)
     for (const auto & var : it.second)
       _depend_vars.insert(var->name());
+
+  if (_displaced_var)
+  {
+    addMooseVariableDependency(_displaced_var);
```

Comment:
Hmmm... this is just not good.  This is adding quite a bit of extra computation.  This will cause the variable value / gradients to get computed _twice_ for every element no matter what.

This isn't your fault... you're doing the best you can... but we can't commit this like this.